analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/CookieShouldBeHttpOnly.cs

```diff
@@ -50,12 +54,40 @@ public sealed class CookieShouldBeHttpOnly : ObjectShouldBeInitializedCorrectlyB
 
         internal CookieShouldBeHttpOnly(IAnalyzerConfiguration analyzerConfiguration) : base(analyzerConfiguration, DiagnosticId, MessageFormat) { }
 
+        protected override Func<bool> DetermineIfDefaultConstructorIsSafe(SonarAnalysisContext context, AnalyzerOptions options)
+        {
+            foreach (var fullPath in context.ProjectConfiguration(options).FilesToAnalyze.FindFiles("web.config"))
+            {
+                var webConfig = File.ReadAllText(fullPath);
+                if (webConfig.Contains("<system.web>") && XmlHelper.ParseXDocument(webConfig) is { } doc
+                    && IsHttpOnlyCookies(doc))
+                {
+                    return () => true;
```

Comment:
I don't see a reason for this lambda indirection. From the point the decision is taken based on current context, it's either `true` or `false`. `()=>true` is the same state information as `true` so we should be able to remove the indirection.